ee/clickhouse/queries/actor_base_query.py

```diff
@@ -0,0 +1,70 @@
+import json
+from typing import Any, Dict, List, Optional, Tuple, TypedDict
+
+from rest_framework import serializers
+
+from ee.clickhouse.client import sync_execute
+from posthog.models import Entity, Filter, Team
+
+
+class ActorResponse(TypedDict):
+    id: str
+    created_at: Optional[str]
+    properties: Dict[str, Any]
+    is_identified: Optional[bool]
+    name: str
+    distinct_ids: List[str]
+
+
+class ActorBaseQuery:
+    aggregating_by_groups = False
+
+    def __init__(self, team: Team, filter: Filter, entity: Optional[Entity] = None):
+        self.team = team
+        self.entity = entity
+        self.filter = filter
+
+        if self.entity and self.entity.math == "unique_group":
```

Comment:
Comment got buried: I don't think this should be handled this way and instead either inlined or extracted into a property.